modin/pandas/groupby.py

```diff
@@ -954,14 +976,17 @@ def _default_to_pandas(self, f, *args, **kwargs):
             and len(self._by.columns) == 1
         ):
             by = self._by.columns[0] if self._drop else self._by.to_pandas().squeeze()
-        elif isinstance(self._by, type(self._query_compiler)):
+        elif self._drop and isinstance(self._by, type(self._query_compiler)):
             by = list(self._by.columns)
         else:
             by = self._by
 
         by = try_cast_to_pandas(by, squeeze=True)
+        by = GroupBy.validate_by(by)
```

Comment:
[`GroupBy.validate_by`](https://github.com/modin-project/modin/blob/4abb53b4a04a66e461a2169f585e474113781be4/modin/data_management/functions/default_methods/groupby_default.py#L34) splits 2D 'by' into a list of 1D Series